packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx

```diff
@@ -65,6 +79,44 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
     onInputChange,
   } = props;
 
+  const _insertBeforeItem = (item: T): void => {
+    const draggedItemIndex = selectedItems.indexOf(draggedItem!);
+    const draggedItemsIndices = focusedItemIndices.includes(draggedItemIndex)
+      ? [...focusedItemIndices]
+      : [draggedItemIndex];
+    const insertIndex = selectedItems.indexOf(item);
+    dropItemsAt(insertIndex, draggedItemsIndices);
+  };
+
+  const _onDragEnter = (item?: any, event?: DragEvent): string => {
+    // return string is the css classes that will be added to the entering element.
+    return dragEnterClass;
+  };
+
+  const _onDrop = (item?: any, event?: DragEvent): void => {
+    if (draggedItem) {
+      _insertBeforeItem(item);
+    }
+  };
+
+  const _onDragStart = (item?: any, itemIndex?: number, selectedItems?: any[], event?: MouseEvent): void => {
```

Comment:
I think for all these functions, we should be having an option to override/achieve behavior of the caller. This way if the caller wants to achieve drag/drop in a different manner than the default, we give them that option. Right now, we aren't giving that option. I would add a check for the props containing any callback on these drag/drop properties, and if they dont exist we can use the default behavior. Some examples where we are already doing this is - getItemsCopyText, _onInputFocus , _onCopy etc.

Reply:
That makes sense, I'll add that

Reply:
So all of these functions are a part of the IDragDropEvents. I'm thinking just make that a prop and use it if there, otherwise use all these functions. I think this makes more sense, because the functionality of all of these functions is intertwined, but am open to other opinions

Reply:
Yea i think that makes sense, as they are dependent on each other.